Mathlib/RingTheory/Int/Basic.lean

```diff
@@ -131,3 +131,20 @@ theorem eq_pow_of_mul_eq_pow_odd {a b c : ℤ} (hab : IsCoprime a b) {k : ℕ} (
   ⟨eq_pow_of_mul_eq_pow_odd_left hab hk h, eq_pow_of_mul_eq_pow_odd_right hab hk h⟩
 
 end Int
+
+namespace Nat
+
+theorem eq_pow_of_mul_eq_pow_left {a b c k : ℕ} (hab : Coprime a b) (h : a * b = c ^ k) :
+  ∃ d, a = d ^ k := by
+  obtain ⟨d, hd⟩ := exists_associated_pow_of_mul_eq_pow (isUnit_iff_eq_one.mpr hab) h
+  use d
+  rw [associated_iff_eq.mp hd]
```

Comment:
This can use [exists_eq_pow_of_mul_eq_pow](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/GCDMonoid/Basic.html#exists_eq_pow_of_mul_eq_pow):
```suggestion
  exists_eq_pow_of_mul_eq_pow (isUnit_iff_eq_one.mpr hab.gcd_eq_one) h
```

But then I wonder if we still need the Nat specialization